app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee

```diff
@@ -75,8 +75,16 @@ ShortFormApplicationController = (
 
   $scope.emailRegex = SharedService.emailRegex
 
-  $scope.trackAutofill = ->
+  $scope.startAutofilledApp = ->
     AnalyticsService.trackFormSuccess('Application', 'Start with these details')
+    # Start page after autofill preview
+    $scope.go($scope.getFirstFillableAppPage())
+
+  $scope.getFirstFillableAppPage = ->
```

Comment:
I noticed that this function already exists in `ShortFormNavigationService` which is already injected in this controller and you can just call to return the right statename...

```
  Service.initialState = () ->
    if ListingIdentityService.isSale(ShortFormApplicationService.listing)
      'dahlia.short-form-application.prerequisites'
    else
      'dahlia.short-form-application.name'
```

Reply:
fixed!